### DISCUSSION TITLE
Should this be part of UnJS?

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [nitrojs/nitro#1222] [General]
Heya! Big fan of Nitro and everything else in the UnJS ecosystem 🎉 

I have a bit of a tricky question that I think is worth discussing here: 

_Is Nitro "appropriate" to be part of the UnJS ecosystem?_

As per the [UnJS governance guidelines](https://github.com/unjs/governance#readme) (emphasis mine):

> UnJS is a collection of **independent** JavaScript libraries, tools, and utilities.
> 
> Each project is usually made for **a specific purpose in the ecosystem** following the [Unix Philisofy](http://www.catb.org/~esr/writings/taoup/html/ch01s06.html) and maintained by its author and designated maintainers.
> 
> **UnJS Goals**
> • High-quality and **single-purpose** javascript utilities
> [...]
> • **Independent development of solutions that benefit the majority of JS ecosystem**

In my opinion, Nitro currently doesn't fit the governance vision of UnJS, as it's not a single-purpose library, but instead a framework built out of a collection of said libraries. 

Possible solutions: 

1. Move Nitro to a dedicated Organization
2. Update UnJS' Governance Guidelines to explicitly allow frameworks comprised of other UnJS projects
3. Tell Rijk to shut up and move on 😄

**Comment by pi0** [comment]
Hi and thanks for the kind words dear @rijkvanzanten 

Nitro is following most of the UnJS design goals. It aims to make agnostic servers with the best effort to keep abstraction and hackability and simplicity. Think of Nitro to UnJS as what Ubuntu is to Unix tools.

It has been something I was thinking about at some point Nitro is to be moved to a dedicated org and team but for the time being, it is easier to maintain both ecosystem and (semi-meta framework) Nitro in the same org. 

> allow frameworks comprised of other UnJS projects

If you have some ideas to improve the goals PR more than welcome. There are other projects such as [c12](https://github.com/unjs/c12) that depends on rest of ecosystem. They are "composables by best efforts".

I hope that solves your questions and best!

**Reply by rijkvanzanten** [reply]
> Nitro is following most of the UnJS design goals. It aims to make agnostic servers with the best effort to keep abstraction and hackability and simplicity. Think of Nitro to UnJS as what Ubuntu is to Unix tools.

I think my personal concern came from the idea that _making an agnostic server_ is a way bigger scope of functionality than any of the other individual (single-purpose) packages currently do. To use your c12 example, that's to load configuration and that's it, whereas Nitro currently tries to solve _a lot_ of different problems at once, including storage abstraction, cache handling, asset generation, and even bundling. 

> It has been something I was thinking about at some point Nitro is to be moved to a dedicated org and team but for the time being, it is easier to maintain both ecosystem and (semi-meta framework) Nitro in the same org.

Makes sense!

> If you have some ideas to improve the goals PR more than welcome. There are other projects such as [c12](https://github.com/unjs/c12) that depends on rest of ecosystem. They are "composables by best efforts".

Thanks! I'll give it some thought and try to formulate a nice way to describe the difference and welcome both. 

> I hope that solves your questions and best!

Yeah! I hope it goes without saying that this "problem" is no more than a thought experiment at best of course, I was just very curious to hear what your take was on the difference between "frameworks" and "libraries" in the context of UnJS (and if you'd even consider Nitro a "framework" as such 😄) Groetjes!
